aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -85,12 +85,19 @@ Tensor& _clamp_min_out_cpu(Tensor& result, const Tensor& self, Scalar min) {
   return legacy::th::_th_clamp_min_out(result, self, min);
 }
 
+Tensor& fill_out(Tensor& self, const Scalar value) {
+  auto iter = TensorIterator::nullary_op(self);
+  fill_stub(iter->device_type(), *iter, value);
+  return self;
+}
+
 Tensor& fill_(Tensor& self, Scalar value) {
-  return at::legacy::th::_th_fill_(self, value);
+  return fill_out(self, value);
 }
 
 Tensor& fill_(Tensor& self, const Tensor& value) {
-  return at::legacy::th::_th_fill_(self, value);
+  AT_ASSERTM(value.dim() == 0, "fill_ only supports 0-dimension value tensor.");
```

Comment:
use `TORCH_CHECK`

The previous error message includes the dimension of value which is sometimes helpful for debugging:

"fill only supports a 0-dimensional value tensor, but got tensor with 1 dimension(s)"